app/compare/diff_preprocessing.tsx

```diff
@@ -0,0 +1,128 @@
+import { invocation as invocation_proto } from "../../proto/invocation_ts_proto";
+import { command_line } from "../../proto/command_line_ts_proto";
+
+/**
+ * A unique token used to identify parts of the diff that were hidden due to pre-processing options.
+ */
+export const HIDDEN_TOKEN = `/__HIDDEN__${Math.random()}/`;
+
+export type PreProcessingOptions = {
+  sortEvents?: boolean;
+  hideTimingData?: boolean;
+  hideConsoleOutput?: boolean;
+  hideInvocationIds?: boolean;
+  hideUuids?: boolean;
+};
+
+export function computeTextForDiff(
```

Comment:
don't love that we do half of these manipulations on the proto itself, and the other half on the rendered json. i get why we're doing it this way, but wish we could just do one or the other